### DISCUSSION TITLE
How to fill a pivot table.

### DISCUSSION THREAD

**Discussion by larsvanherwijnen** [filamentphp/filament#7052] [Help]
I am trying to save data to a pivot table between characteristic and product (characteristic_product).

I have a productresource for creating and edit a product. A product has characteristic those characteristic assigned via the category the product is in.

The characteristic_product also stores the value of that characteristic the columns in the characteristic_product table are characteristic_id, product_id and value.
Now the problem am having is that I don't know to fill the pivot table.

What is have already tried is changing the handleRecordCreation and handleRecordUpdate. And looping through the characteristics array that tought would be in the data array. But is not the case.
![image](https://github.com/filamentphp/filament/assets/77294984/17594e14-461e-49d6-902e-cb5140982340)

I would appreciate any guidance on how in can handle this.

Thanks!
How i build my form:
![image](https://github.com/filamentphp/filament/assets/77294984/98a6d75c-ec94-433d-b991-8b156bcce21e)

**Comment by pboivin** [comment]
Not 100% sure I understand the context but have you tried introducing a pivot model? (e.g. `ProductCharacteristic`). You could rewrite the relationship to a `HasMany` (Product has-many ProductCharacteristic) and write to the pivot attributes directly.

**Reply by lukacavic** [reply]
@larsvanherwijnen I have same problem, did you solve it?
